metagpt/actions/rebuild_sequence_view.py

```diff
@@ -8,49 +8,293 @@
+                'The functional descriptions of Actors and Use Cases in the generated UML 2.0 Use Case must not conflict with the information in "Mermaid Class Views".',
+                # 'Only steps that involve input, output, and interactive operations with the external system at the same time can be considered as independent use cases.',
+                # "Only steps that involve input, output, and interactive operations with the external system at the same time can be considered as independent use cases, steps that do not meet any one condition should be incorporated into other use cases.",
+                'The section under `if __name__ == "__main__":` of "Source Code" contains information about external system interactions with the internal system.',
+                "Return a markdown JSON object with:\n"
+                '- a "description" key to explain what the whole source code want to do;\n'
+                '- a "use_cases" key list all use cases, each use case in the list should including a `description` key describes about what the use case to do, a `inputs` key lists the input names of the use case from external sources, a `outputs` key lists the output names of the use case to external sources, a `actors` key lists the participant actors of the use case, a `steps` key lists the steps about how the use case works step by step, a `reason` key explaining under what circumstances would the external system execute this use case.\n'
+                '- a "relationship" key lists all the descriptions of relationship among these use cases.\n',
+            ],
+        )
+
+        code_blocks = parse_json_code_block(rsp)
+        for block in code_blocks:
+            detail = SQVUseCaseDetails.model_validate_json(block)
+            await self.graph_db.insert(
+                subject=ns_class_name, predicate=GraphKeyword.HAS_CLASS_USE_CASE, object_=detail.model_dump_json()
+            )
+
+    @retry(
+        wait=wait_random_exponential(min=1, max=20),
+        stop=stop_after_attempt(6),
+        after=general_after_log(logger),
+    )
+    async def _rebuild_sequence_view(self, ns_class_name):
+        await self._rebuild_use_case(ns_class_name)
+
+        prompts_blocks = []
+        use_case_markdown = await self._get_class_use_cases(ns_class_name)
+        if not use_case_markdown:  # external class
+            await self.graph_db.insert(subject=ns_class_name, predicate=GraphKeyword.HAS_SEQUENCE_VIEW, object_="")
             return
-        content = await aread(filename=src_filename, encoding="utf-8")
-        content = f"```python\n{content}\n```\n\n---\nTranslate the code above into Mermaid Sequence Diagram."
-        data = await self.llm.aask(
-            msg=content, system_msgs=["You are a python code to Mermaid Sequence Diagram translator in function detail"]
+        block = f"## Use Cases\n{use_case_markdown}"
```

Comment:
seems there are multi parts of prompt_blocks, can these to a uniform logic which passes some params and return prompt.
like Line 215-232 98-111  164-177

Reply:
These are details of data processing, involving different input data assembly and various system messages. There are no cases with identical inputs or identical system messages, so there is no further encapsulation. `llm.aask` is already at the maximum granularity.